### ISSUE TITLE
Javascript i18n issue

### ISSUE THREAD

**Issue by shenhf** [TrestleAdmin/trestle#162]
```
# app/views/layouts/trestle/admin.html.erb
   ...
    <script type="text/javascript">
      <% Trestle.config.javascript_i18n_keys.each do |key| %>
      Trestle.i18n['<%= key %>'] = "<%= escape_javascript(t(key, default: t(key, locale: :en))) %>";
      <% end %>

      Trestle.localize(<%= i18n_fallbacks.map { |l| "'#{l}'" }.join(", ").html_safe %>);
    </script>
```
@spohlenz I18n translation text was loaded after confirmation dialog init. So translation will never happen.
I think it should be after app/assets/javascripts/trestle/admin.js.

**Comment by shenhf** [comment]
Since the lib/trestle/configuration.rb define the key in trestle attribute:
```
    # List of i18n keys to pass into the Trestle.i18n JavaScript object
    option :javascript_i18n_keys, [
      "trestle.confirmation.title", "trestle.confirmation.delete", "trestle.confirmation.cancel", "trestle.dialog.error",
      "admin.buttons.ok", "admin.datepicker.formats.date", "admin.datepicker.formats.datetime", "admin.datepicker.formats.time"
    ]
```
Then in app/assets/javascripts/trestle/components/_confirmation.js:
```
Trestle.i18n['admin.confirmation.****']
```
should also changed to: 
```
Trestle.i18n['trestle.confirmation.***']
```

Thanks for the great trestle gem!

**Comment by kaikuchn** [comment]
I'd like to piggyback onto this issue that inline Javascript violates CSP unless one allows for unsafe-inline javascript or uses a hash (currently if only `en` is available: `script-src 'sha256-O9k/snS97p33jn9W9l13gkQCLAO9tQvyMNcQ75mvsZQ'`).

**Comment by simonc** [comment]
Hi 👋  I'd also like to add that if your app is not enabling the `:en` locale then the default breaks with a `I18n::InvalidLocale` error.

**Comment by nguyentrongtam** [comment]
FYI:  I override file `app/assets/javascripts/trestle/components/_confirmation.js` like:
```
Trestle.ready(function() {
  Trestle.confirmation = {};

  Trestle.confirmation.DEFAULTS = {
    rootSelector:   'body',
    singleton:      true,
    popout:         true,
    title:          Trestle.i18n['trestle.confirmation.title'] || 'Are you sure?',
    btnOkIcon:      '',
    btnOkClass:     'btn-primary',
    btnOkLabel:     Trestle.i18n['trestle.confirmation.ok'] || 'OK',
    btnCancelIcon:  '',
    btnCancelClass: 'btn-default',
    btnCancelLabel: Trestle.i18n['trestle.confirmation.cancel'] || 'Cancel',
    copyAttributes: ''
  };

  Trestle.confirmation.CONFIRM = $.extend({}, Trestle.confirmation.DEFAULTS, {
    selector: '[data-toggle="confirm"]'
  });

  Trestle.confirmation.DELETE = $.extend({}, Trestle.confirmation.DEFAULTS, {
    selector:   '[data-toggle="confirm-delete"]',
    btnOkClass: 'btn-danger',
    btnOkLabel: Trestle.i18n['trestle.confirmation.delete'] || ' Delete'
  });

  // This must be bound to an element beneath document so
  // that it is fired before any jquery_ujs events.
  var root = $('body');
  // Ensure it is only initialized once
  if (root.data('bs.confirmation')) return;

  // Delete confirmation
  new $.fn.confirmation.Constructor(root, Trestle.confirmation.DELETE);

  // General confirmation
  new $.fn.confirmation.Constructor(root, Trestle.confirmation.CONFIRM);
});

```
And it's working.

**Comment by spohlenz** [comment]
I've now merged #175 which fixes the original issue.

> I'd like to piggyback onto this issue that inline Javascript violates CSP unless one allows for unsafe-inline javascript or uses a hash (currently if only en is available: script-src 'sha256-O9k/snS97p33jn9W9l13gkQCLAO9tQvyMNcQ75mvsZQ').

This should now be fixed in 85b1f81507d4f1ce38dac60f0456d1c155102d0c.

> Hi 👋 I'd also like to add that if your app is not enabling the :en locale then the default breaks with a I18n::InvalidLocale error.

This has also now been fixed in 2540af33dda8c056d0f0e846bdfc894c3a538a8a.

Thank you all for reporting these issues. I'll be putting out a new release shortly.
